src/IssueViz/Editor/QuickActions/QuickFixes/QuickFixActionsSource.cs

```diff
@@ -88,19 +89,20 @@ public async Task<bool> HasSuggestedActionsAsync(ISuggestedActionCategorySet req
         {
             var hasActions = false;
 
-            await threadHandling.RunOnUIThread(() => hasActions = IsOnIssueWithQuickFixes(range, out _));
+            await threadHandling.RunOnUIThread(() => hasActions = IsOnIssueWithApplicableQuickFixes(range, out _));
 
             return hasActions;
         }
 
-        private bool IsOnIssueWithQuickFixes(SnapshotSpan range, out IEnumerable<IAnalysisIssueVisualization> issuesWithFixes)
+        private bool IsOnIssueWithApplicableQuickFixes(SnapshotSpan range, out IEnumerable<IAnalysisIssueVisualization> issuesWithFixes)
         {
             var tagSpans = issueLocationsTagAggregator.GetTags(range);
 
             issuesWithFixes = tagSpans
                 .Select(x => x.Tag.Location)
                 .OfType<IAnalysisIssueVisualization>()
-                .Where(x => x.QuickFixes.Any());
+                .Where(x =>
+                    x.QuickFixes.Any(fix => fix.IsApplicable(textView.TextSnapshot)));
```

Comment:
There is a potential performance hit here.
For every quick fix, we will check 3 times if it's still applicable:
1. Once by VS to know if it should render the light bulb
2. Another time when the user opens the lightbulb to populate the list of available actions
3.  A third time when the quick fix is applied

So 3 times we will do a span translation and string comparison of all the edits, of all the fixes that an issue has.

I don't know if it's expensive or not. I didn't notice a performance hit but I was testing on a small class that returned a small amount of issues, with very few fixes. Just something to keep in mind that we might need to optimize this.